core/src/main/java/org/apache/calcite/sql/SqlCallBinding.java

```diff
@@ -145,10 +144,11 @@ public SqlCall getCall() {
       }
       final SqlOperandCountRange range = checker.getOperandCountRange();
       final List<SqlNode> list = Lists.newArrayList(operandList);
+      final SqlCall defaultCall = createDefaultCall();
       while (list.size() < range.getMax()
           && checker.isOptional(list.size())
           && checker.isFixedParameters()) {
-        list.add(DEFAULT_CALL);
+        list.add(defaultCall);
```

Comment:
Suggest you don’t create the defaultCall outside the loop. Most calls do not need default arguments. 

You could in-line the defaultCall method, because it’s very simple.

Reply:
Thanks, I was not aware that most case indeed did not require the default call. Changed.